### DISCUSSION TITLE
Invalid active developer path

### DISCUSSION THREAD

**Discussion by tqrahman** [Code-and-Response/ClusterDuck-Protocol#330] [QuackOverflow] [answered]
I am using macOS and after I installed macOS Sonoma, I am receiving this error when I am trying to flash using Arduino: 

> error xcrun: error: invalid active developer path (/Library/Developer/CommandLineTools), missing xcrun at: /Library/Developer/CommandLineTools/usr/bin/xcrun

**Comment by tqrahman** [comment, marked as answer]
Ok the problem was that every time we upgrade to a new macOS, we have to reinstall [Xcode](https://developer.apple.com/support/xcode/). 

After I reinstalled Xcode, the error went away.

**Comment by knouse1344** [comment]
updated xcode, that seemed to solve something but now a new problem:  

`/Users/bryanknouse/Library/Arduino15/packages/esp32/tools/xtensa-esp32-elf-gcc/esp-2021r2-patch5-8.4.0/bin/../lib/gcc/xtensa-esp32-elf/8.4.0/../../../../xtensa-esp32-elf/bin/ld: libraries/ESPAsyncWebServer/WebAuthentication.cpp.o: in function getMD5(unsigned char*, unsigned short, char*)':
/Users/bryanknouse/Documents/Arduino/libraries/ESPAsyncWebServer/src/WebAuthentication.cpp:73: undefined reference to mbedtls_md5_starts' collect2: error: ld returned 1 exit status exit status 1
Error compiling for board T-Beam.`

![image](https://github.com/Call-for-Code/ClusterDuck-Protocol/assets/1772065/c2fd3ec3-766c-43da-9930-54813266a808)

**Comment by knouse1344** [comment]
Went in and had to accept Xcode license agreement, and also switched to main branch.    Still having the issue.  
![image (3)](https://github.com/Call-for-Code/ClusterDuck-Protocol/assets/1772065/970da7b7-b6f3-4f5d-a5ff-71166dd27ad8)
![image (2)](https://github.com/Call-for-Code/ClusterDuck-Protocol/assets/1772065/ca4e0479-b9ac-4be8-8e2d-2b7915e82107)

Found that ESPAsyncWebServer uses a different name when downloading on github as a zip versus thru Arduino library manager.

The [Github uses](https://github.com/me-no-dev/ESPAsyncWebServer) name "ESPAsyncWebServer.h" and Arduino Library Manager installs "ESPAsyncWebSrv.h" which causes issues if you use library manager, because CDP uses the former name scheme.

Re-installed latest library by downloading Zip from ESPAsyncWebServer github.  We good now.

![image](https://github.com/Call-for-Code/ClusterDuck-Protocol/assets/1772065/174cf9f8-29a7-481e-8a5c-e898eb18d458)
